### ISSUE TITLE
Specify the fragment parsing algorithm

### ISSUE THREAD

**Issue by nox** [Ygg01/xml5ever#23]
Cc @annevk

**Comment by Ygg01** [comment]
Hi nox, I assume you're interested in supporting following features: [serializing XHTML]( https://html.spec.whatwg.org/multipage/xhtml.html#serialising-xhtml-fragments) and [parsing XHTML](https://html.spec.whatwg.org/multipage/xhtml.html#serialising-xhtml-fragments)

**Comment by nox** [comment]
Yes, and I think the parsing one doesn't really fit with the more lax XML5 rules.

**Comment by annevk** [comment]
Other than step 5 of https://html.spec.whatwg.org/multipage/xhtml.html#xml-fragment-parsing-algorithm what is wrong?

**Comment by nox** [comment]
@annevk Nothing, I guess. I just fancy the idea of having a context element better.

**Comment by Ygg01** [comment]
I do have a large update in the works, that has been delayed by some RL issues, but once that is done. I'll be happy to work/help on fragment parsing.

**Comment by nox** [comment]
@Ygg01 More or less I would like to avoid having to feed x5e actual strings for the open and close tags of the context element.

**Comment by Ygg01** [comment]
@nox So I assume like [this line](https://github.com/servo/html5ever/blob/73294d45420a974ecfea065ccfe521f573de83fd/src/driver.rs#L55) from h5e?

**Comment by nox** [comment]
I think we just need a way to give the initial list of namespaces in scope, instead of the tag name and the attributes.

**Comment by Ygg01** [comment]
@nox I tried reaching out to IRC, so I'll ask question here. Are there any examples of xhtml parsing I could use to construct test cases?

@annevk @nox Right now, my I'm creating a context element not as string input but as a node (like in html5ever). So I assume this will differ than https://html.spec.whatwg.org/multipage/xhtml.html#xml-fragment-parsing-algorithm (step 6).  

If I understand correctly, if you have context element `context` and input string as following `<a></context><test><context>` it would construct a different tree (or . Is such behavior desired?
